common/test/acceptance/tests/studio/test_studio_settings.py

```diff
@@ -290,3 +290,9 @@ def set_wrong_inputs_to_fields(self):
                 "Course Announcement Date": '"string"',
             }
         )
+
+    def test_hidden_fields_not_displayed(self):
```

Comment:
I'm not sure this test is really useful (since you have to update the code 2 places, and it won't catch things that you forgot to filter out).

I think what would be more useful is a test that checks for the elements that ARE present (and verifies no extra settings are present). That way, if you add something new, the test fails and you have to think about whether or not it should be exposed in Advanced Settings. If it should be, you update the test. This would have caught a number of items that inadvertently showed up in Advanced Settings.